opennms-rrd/opennms-rrd-rrdtool/opennms-rrdtool-api/src/main/java/org/opennms/netmgt/rrd/rrdtool/AbstractJniRrdStrategy.java

```diff
@@ -0,0 +1,210 @@
+        String[] commandArray = StringUtils.createCommandArray(command);
+        Process process;
+        try {
+             process = Runtime.getRuntime().exec(commandArray, null, workDir);
+        } catch (IOException e) {
+            IOException newE = new IOException("IOException thrown while executing command '" + command + "' in " + workDir.getAbsolutePath() + ": " + e);
+            newE.initCause(e);
+            throw newE;
+        }
+        
+        // this closes the stream when its finished
+        byte[] byteArray = FileCopyUtils.copyToByteArray(process.getInputStream());
+        
+        // this close the stream when its finished
+        String errors = FileCopyUtils.copyToString(new InputStreamReader(process.getErrorStream()));
+        
+        // one particular warning message that originates in libart should be ignored
+        if (errors.length() > 0 && errors.contains(IGNORABLE_LIBART_WARNING_STRING)) {
+        	LOG.debug("Ignoring libart warning message in rrdtool stderr stream: {}", IGNORABLE_LIBART_WARNING_STRING);
+        	errors = errors.replaceAll(IGNORABLE_LIBART_WARNING_REGEX, "");
+        }
+        if (errors.length() > 0) {
+            throw new RrdException(errors);
+        }
+        return byteArray;
+    }
+
+    /**
+     * No stats are kept for this implementation.
+     *
+     * @return a {@link java.lang.String} object.
+     */
+    @Override
+    public String getStats() {
+        return "";
+    }
+
+    /**
+     * <p>getGraphLeftOffset</p>
+     *
+     * @return a int.
+     */
+    @Override
+    public int getGraphLeftOffset() {
+        return 0;
+    }
+    
+    /**
+     * <p>getGraphRightOffset</p>
+     *
+     * @return a int.
+     */
+    @Override
+    public int getGraphRightOffset() {
+        return 0;
+    }
+
+    /**
+     * <p>getGraphTopOffsetWithText</p>
+     *
+     * @return a int.
+     */
+    @Override
+    public int getGraphTopOffsetWithText() {
+        return 0;
+    }
+
+    /**
+     * <p>getDefaultFileExtension</p>
+     *
+     * @return a {@link java.lang.String} object.
+     */
+    @Override
+    public String getDefaultFileExtension() {
+        return ".rrd";
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public RrdGraphDetails createGraphReturnDetails(String command, File workDir) throws IOException, org.opennms.netmgt.rrd.RrdException {
+        // Creating Temp PNG File
+        File pngFile = File.createTempFile("opennms.rrdtool.", ".png");
+        command = command.replaceFirst("graph - ", "graph " + pngFile.getAbsolutePath() + " ");
+
+        int width;
+        int height;
+        String[] printLines;
+        InputStream pngStream;
+
+        try {
+            // Executing RRD Command
+            InputStream is = createGraph(command, workDir);
+            
+            // Processing Command Output
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+            
```

Comment:
You can use Java 7 try-resource-blocks.

    try (
         InputStream is = ....; 
         BufferedReader reader = ....) {
                reader.read(...);
    }

Then you do not have to close it manually.